Refresh.GameServer/Types/Assets/AssetFlags.cs

```diff
@@ -0,0 +1,114 @@
+namespace Refresh.GameServer.Types.Assets;
+
+[Flags]
+public enum AssetFlags
+{
+    None = 0,
+    /// <summary>
+    /// This asset can be dangerous to end users.
+    /// </summary>
+    Dangerous = 1 << 0,
+    /// <summary>
+    /// This asset is a media-type asset, e.g. a PNG or TEX.
+    /// </summary>
+    Media = 1 << 1,
+    /// <summary>
+    /// This asset will only ever be created by mods.
+    /// </summary>
+    Modded = 1 << 2,
+}
+
+public static class AssetSafetyLevelExtensions
+{
+    public static AssetFlags FromAssetType(GameAssetType type, GameSerializationMethod? method)
+    {
+        return type switch
+        {
+            // Non-binary levels will not be loadable 
+            GameAssetType.Level => method == GameSerializationMethod.Binary
+                ? AssetFlags.None
+                : AssetFlags.Dangerous | AssetFlags.Modded,
+            GameAssetType.StreamingLevelChunk => method == GameSerializationMethod.Binary
+                ? AssetFlags.None
+                : AssetFlags.Dangerous | AssetFlags.Modded,
+            GameAssetType.Plan => method == GameSerializationMethod.Binary ? AssetFlags.None : AssetFlags.Modded,
```

Comment:
Unsure about this "if not binary, then modded" classification. I suppose it's true that non-binary files being uploaded usually means an untrustworthy origin but if the game deserializes it the same, it should be treated the same. Not to mention code quality.

Reply:
that was something i was unsure about aswell, it depends on how precise we want to classify modded assets (since we want to provide a "is modded" flag for levels)

Reply:
I'm leaning towards the slightly less complicated solution of just treating all the types as the same, unless we can somehow make this less annoying to follow/work with.

Reply:
> I'm leaning towards the slightly less complicated solution of just treating all the types as the same, unless we can somehow make this less annoying to follow/work with.

I'm fine with treating them all the same, with the exception of unloadable ones (like `LVLt` or `TEXt`), those should definitely specified as "dangerous" in my opinion

perhaps it could be refactored to something like
```
AssetFlags flags = type switch {
	// All these branches are changed to ignore the serialization method
	GameAssetType.Level => AssetFlags.None,
};

//After we get the initial flags, we append on more flags for all of the unloadable ones, like `LVLt`, or `TXTt`
flags |= type switch {
	GameAssetType.Level => method == Binary ? AssetFlags.None : AssetFlags.Dangerous,
	GameAssetType.Texture => method == CompressedTexture ? AssetFlags.None : AssetFlags.Dangerous,
	else => AssetFlags.None,
};

return flags;
```

Reply:
Yeah, I prefer that actually.